src/entry.rs

```diff
@@ -635,11 +642,87 @@ impl<'a> EntryFields<'a> {
         if let Ok(mode) = self.header.mode() {
             set_perms(dst, Some(&mut f), mode, self.preserve_permissions)?;
         }
+        if self.preserve_ownerships {
+            set_ownerships(
+                dst,
+                Some(&mut f),
+                self.header.uid().ok(),
+                self.header.gid().ok(),
+            )?;
+        }
```

Comment:
This pattern of "set all the things and metadata" happens in a few places in this function, could they be refactored where one function "sets all the things" and internally it dispatches to ownership, permission, mtime, etc?

Reply:
Sure.